chat.go

```diff
@@ -34,12 +36,37 @@ type ChatCompletionMessage struct {
 	FunctionCall *FunctionCall `json:"function_call,omitempty"`
 }
 
+var (
+	FunctionCallNone = &FunctionCall{}
+	FunctionCallAuto = &FunctionCall{}
+)
+
 type FunctionCall struct {
 	Name string `json:"name,omitempty"`
 	// call function with arguments in JSON format
 	Arguments string `json:"arguments,omitempty"`
 }
 
+func (f *FunctionCall) MarshalJSON() ([]byte, error) {
+	switch f {
+	case FunctionCallNone:
+		return []byte(`"none"`), nil
+	case FunctionCallAuto:
+		return []byte(`"auto"`), nil
+	default:
+	}
+	type functionCallAlias FunctionCall
+	return json.Marshal((*functionCallAlias)(f))
```

Comment:
Developers unfamiliar with Go may be a bit confused, so please add a comment that the use of type aliases prevents recursive calls.

Reply:
Learned something from this comment 👍. It would be helpful to clarify this is a type definition rather than an alias. In go's terms, an alias is an identifier bound to a type so it receives all the methods from it too.